build_dependency_tree.py

```diff
@@ -3,14 +3,27 @@
 import sys
 import re
 from pathlib import Path
+from typing import Optional
 from typing import List
+from typing import Union
 
-args = [arg for arg in sys.argv]
-files_in_dir = [
-    [file for file in Path(root).glob('*') # TODO: Make recursive search 
-     if file.suffix in ('h', 'hpp', 'c', 'cpp')]
-    for root in args
-]
+args: List[Union[bytes, str]] = [arg for arg in sys.argv]
+files_in_project = []
```

Comment:
instead of having a globally defined variable which is just needed by one function (it seems), we pass a default empty list in the function